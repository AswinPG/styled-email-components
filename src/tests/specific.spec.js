import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import styled from '../';

const render = element => ReactTestRenderer.create(element).toJSON();

describe('specific', () => {
  it('supports mail-specific props', () => {
    const MsoComponent = styled.center`
      mso-hide: all;
    `;
    expect(render(<MsoComponent />)).toMatchSnapshot();
  });

  it('supports vml specific elements', () => {
    const AnchorLock = styled('w:anchorlock')``;
    expect(render(<AnchorLock />)).toMatchSnapshot();
  });

  fit('supports custom attributes', () => {
    const RoundRect = styled.vml.roundrect.attrs({
      'xmlns:v': 'urn:schemas-microsoft-com:vml',
      'xmlns:w': 'urn:schemas-microsoft-com:office:word',
    })``;
    expect(render(<RoundRect />)).toMatchSnapshot();
  });

  it('outputs correct units', () => {
    const Box = styled.center`
      padding: 10px;
      margin: 10in;
      left: -1em;
      right: 5pt;
      line-height: 1.2;
    `;
    expect(render(<Box />)).toMatchSnapshot();
  });

  it('expands color to 6 digits', () => {
    const Color = styled.p`
      color: #000;
      border: 1px solid #333;
    `;
    expect(render(<Color />)).toMatchSnapshot();
  });
});
