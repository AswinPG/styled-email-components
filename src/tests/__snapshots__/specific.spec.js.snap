// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`specific expands color to 6 digits 1`] = `
<p
  style={
    Object {
      "borderColor": "#333333",
      "borderStyle": "solid",
      "borderWidth": "1px",
      "color": "#000000",
    }
  }
/>
`;

exports[`specific outputs correct units 1`] = `
<center
  style={
    Object {
      "left": "-1em",
      "marginBottom": "10in",
      "marginLeft": "10in",
      "marginRight": "10in",
      "marginTop": "10in",
      "paddingBottom": "10px",
      "paddingLeft": "10px",
      "paddingRight": "10px",
      "paddingTop": "10px",
      "right": "5pt",
    }
  }
/>
`;

exports[`specific supports mail-specific props 1`] = `
<center
  style={
    Object {
      "msoHide": "all",
    }
  }
/>
`;

exports[`specific supports vml specific elements 1`] = `
<w:anchorlock
  style={Object {}}
/>
`;
